### ISSUE TITLE
HaplotypeCaller -contamination argument does not affect AD, but should

### ISSUE THREAD

**Issue by droazen** [broadinstitute/gatk#4412]
The `-contamination` argument in `HaplotypeCaller` currently affects the PLs, but not AD, which seems bizarre. @ldgauthier agrees that it should affect AD as well. This is an issue in both GATK3 and GATK4.

**Comment by eitanbanks** [comment]
@ldgauthier I believe that we used to adjust the ADs and then it confused the hell out of people.  They could see those bases in the reads so complained that the ADs were wrong.  FWIW I don't think that the PLs and ADs need to "match" -- the user is asking us to take contamination into account when calculating the likelihoods, but that doesn't mean that the bases aren't present (nor that those bases are necessarily contamination).

**Comment by ldgauthier** [comment]
Fair point.  We don't drop 1 read out of every thousand in a exome pileup because they're base quality 30.

**Comment by eitanbanks** [comment]
Not sure that's right.  Does it include bases from MQ0 reads?

**Comment by davidbenjamin** [comment]
I'm for keeping all the bases in the AD because what the `-contamination` argument does is very strict.  The adjusted alt AD would be a conservative lower bound and in this case I feel the raw AD would be more informative.  But why not compromise and annotate the variant with the number of reads dropped by `-contamination`?
